staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -314,3 +318,40 @@ func (e *eventBroadcasterImpl) StartRecordingToSink(stopCh <-chan struct{}) {
 		stopWatcher()
 	}()
 }
+
+type eventClientImpl struct {
+	CoreClient         typedv1core.EventsGetter
+	CoreBroadcaster    record.EventBroadcaster
+	V1beta1Client      typedv1beta1.EventsGetter
+	V1beta1Broadcaster EventBroadcaster
+}
+
+// NewEventClient Creates a new event broadcaster.
+func NewEventClient(client clientset.Interface) EventClient {
+	eventClient := &eventClientImpl{}
+	if _, err := client.Discovery().ServerResourcesForGroupVersion(v1beta1.SchemeGroupVersion.String()); err == nil {
+		eventClient.V1beta1Client = client.EventsV1beta1()
+		eventClient.V1beta1Broadcaster = NewBroadcaster(&EventSinkImpl{Interface: eventClient.V1beta1Client.Events("")})
+	} else {
+		eventClient.CoreClient = client.CoreV1()
+		eventClient.CoreBroadcaster = record.NewBroadcaster()
```

Comment:
Given you need CoreBroadcaster for legacy usecases (e.g. leaderelection), let's make the following:

- set coreClient unconditionally
- in StartRecordingToSink remove else and just check for v1beta1Broadcaster and coreBroadcaster to check whether to start them or not

Reply:
Done.